server/emails/templates/BaseEmail.tsx

```diff
@@ -86,6 +88,28 @@ export default abstract class BaseEmail<T extends EmailProps, S = any> {
       });
       throw err;
     }
+
+    const { metadata } = { ...this.props };
+
+    if (metadata?.notificationId) {
+      try {
+        await Notification.update(
+          {
+            emailedAt: new Date(),
+          },
+          {
+            where: {
+              id: metadata.notificationId,
+            },
+          }
+        );
+      } catch (err) {
+        Logger.error(
+          `Failed to update notification(id=${metadata.notificationId})`,
+          err
+        );
```

Comment:
```suggestion
        Logger.error(
          `Failed to update notification`,
          err,
          metadata,
        );
```

You can pass structured data to be stored with the error log as third argument 👍